tests/system_tests/test_artifacts/test_wandb_artifacts_full.py

```diff
@@ -497,6 +497,15 @@ def test_artifact_wait_failure(wandb_init, timeout):
     run.finish()
 
 
+def test_finished_run_automatically_awaits_artifact(wandb_init):
+    """Check that finishing a run automatically calls `Artifact.wait()` on all artifacts logged in that run."""
+    with wandb_init() as run:
+        artifact = wandb.Artifact("art", type="dataset")
+        run.log_artifact(artifact)
```

Comment:
Small artifacts log so quickly that I'm not sure this is a reliable test (I suspect it works often, but but would fairly often pass without the fix as well). Might be worth mocking or monkeypatching something in to ensure that `.finish()` is the one calling wait and somehow forcing that.